mtgjson5/set_builder.py

```diff
@@ -1185,6 +1188,36 @@ def add_other_face_ids(cards_to_act_on: List[MtgjsonCardObject]) -> None:
     LOGGER.info("Finished adding otherFaceIds to group")
 
 
+def link_same_card_different_details(mtgjson_set: MtgjsonSetObject) -> None:
+    """
+    In several Magic sets, the foil and non-foil printings have different text
+    (See 10th Edition, for example). If that's the case, we will link the
+    Foil and NonFoil versions together in the identifiers for easier user management
+    :param mtgjson_set: MTGJSON Set
+    """
+    LOGGER.info(f"Linking multiple printings for {mtgjson_set.code}")
+    cards_seen = {}
+
+    for mtgjson_card in mtgjson_set.cards:
+        if mtgjson_card.identifiers.scryfall_illustration_id not in cards_seen:
+            cards_seen[mtgjson_card.identifiers.scryfall_illustration_id] = mtgjson_card
+            continue
+
+        other_mtgjson_card = cards_seen[
+            mtgjson_card.identifiers.scryfall_illustration_id
+        ]
+        if "nonfoil" in mtgjson_card.finishes:
```

Comment:
should you check that nonfoil is the only one too?

Reply:
For the time being, it's not necessary as all of the cases are a foil vs non-foil discrepancy, which lead to the splitting